src/expr.rs

```diff
@@ -37,6 +41,14 @@ use crate::utils::{
 use crate::vertical::rewrite_with_alignment;
 use crate::visitor::FmtVisitor;
 
+lazy_static! {
+    // This regex may accept invalid float literals (such as `1`, `_` or `2.e3`). That's ok.
+    // We only use it to parse literals whose validity has already been established.
+    static ref FLOAT_LITERAL: Regex =
+        Regex::new(r"^([0-9_]+)(?:\.([0-9_]+)?)?([eE][+-]?[0-9_]+)?$").unwrap();
+    static ref ZERO_LITERAL: Regex = Regex::new(r"^[0_]+$").unwrap();
+}
```

Comment:
Lets add unit tests for this regex to highlight some of correct and incorrect literals that it parses.

Reply:
I've factored out a function and added some tests.